pmd-core/src/test/java/net/sourceforge/pmd/cli/CoreCliTest.java

```diff
@@ -195,19 +195,22 @@ public void testRelativeReportFileLongOption() throws IOException {
 
     @Test
     public void debugLogging() {
-        runPmdSuccessfully("--debug", "--no-cache", "--dir", srcDir, "--rulesets", DUMMY_RULESET);
+        Path reportFile = tempRoot().resolve("out/reportFile.txt");
+        runPmdSuccessfully("--debug", "--no-cache", "--dir", srcDir, "--rulesets", DUMMY_RULESET, "-r", reportFile);
```

Comment:
I've added a report output file as otherwise we report to stdout.
And we close the report files afterwards - which is we close stdout.
We have a outStreamCaptor active, however, the close is delegated to the original stdout stream.
Closing stdout in a unit test is not good for IntelliJ IDEA, as IntelliJ IDEA uses this channel to communicate test progress etc...

Reply:
I think we should never close stdout,,, We should use something similar to https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/output/CloseShieldWriter.html

Reply:
I've undone my changes to CoreCliTest and verified, that #4032 actually fixes it.